### PR TITLE
Fix #17873 - Disable media router by default

### PR DESCRIPTION
**Pull request by jumde**
Resolves https://github.com/brave/brave-browser/issues/17873

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Open brave browser with a clean profile.
2. Open the hamburger menu and verify that cast is not available
3. Verify that SSDP requests are not initiated on start
4. Navigate to brave://settings/extensions - Enable `Media Router` and restart
5. Verify that cast is available and functional.

### CODE REVIEW

File: chromium_src/chrome/browser/media/router/media_router_feature.cc [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+#define MediaRouterEnabled MediaRouterEnabled_ChromiumImpl
+#include "../../../../../../chrome/browser/media/router/media_router_feature.cc"
+#undef MediaRouterEnabled
+
+namespace media_router {
+
+bool MediaRouterEnabled(content::BrowserContext* context) {
+#if !defined(OS_ANDROID)
+  if (!base::FeatureList::GetInstance()->IsFeatureOverridden(
```
Comment (mariospr) [inline chromium_src/chrome/browser/media/router/media_router_feature.cc:19]: Nit: Maybe a small comment to clarify that this is done to override the default could be useful.

**Review by mariospr — APPROVED** [review]
LGTM! Also, tested locally and can confirm that no SSDP requests to 239.255.255.250:1900 are made when the feature is left at its default value, or disabled, and that the Cast option is not shown either. All those things work only when the feature is explicitly enabled now.

File: browser/resources/settings/brave_default_extensions_page/brave_default_extensions_page.html [hunk 2]
```diff
@@ -85,6 +85,18 @@
...
         on-settings-boolean-control-change="onHangoutsEnabledChange_">
     </settings-toggle-button>
+    <settings-toggle-button id="mediaRouterEnabled"
+        class="cr-row"
+        pref="{{prefs.media_router.enable_media_router}}"
+        label="Media Router"
+        sub-label="$i18n{mediaRouterEnabledDesc}">
+        <template is="dom-if" if="[[shouldShowRestartForMediaRouter_(
+          prefs.media_router.enable_media_router.value)]]">
+        <cr-button on-click="restartBrowser_" slot="more-actions">
```
Comment (petemill) [inline browser/resources/settings/brave_default_extensions_page/brave_default_extensions_page.html:95]: formatting, needs more indent

Comment (jumde) [reply browser/resources/settings/brave_default_extensions_page/brave_default_extensions_page.html]: fixed!

**Review by petemill — APPROVED** [review]

**Review by mariospr — APPROVED** [review]
LGTM!

**Comment by stephendonner** [comment]
Verified `PASSED` using 


Brave | 1.31.28 Chromium: 93.0.4577.63 (Official Build) nightly (x86_64)
-- | --
Revision | ff5c0da2ec0adeaed5550e6c7e98417dac77d98a-refs/branch-heads/4577@{#1135}
OS | macOS Version 11.5.2 (Build 20G95)


Steps:
1. clean profile
2. launched Charles Proxy
3. launched Brave
4. confirmed that the `Cast...` menu was not present
Screen Shot 2021-09-08 at 3 45 08 PM

5. using Charles Proxy, verified that SSDP requests are not initiated on start
Screen Shot 2021-09-08 at 3 47 42 PM

6. loaded `brave://settings/extensions` and set `Enable Media Router` to `on` / `enabled`
Screen Shot 2021-09-08 at 3 49 09 PM

7. confirmed that the `Cast...` menu item returned
Screen Shot 2021-09-08 at 3 49 20 PM

8. loaded `youtube.com`, chose a video, and casted to my personal TV
Screen Shot 2021-09-08 at 3 50 23 PM

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 10 x64` using the following build:

```
Brave | 1.31.33 Chromium: 93.0.4577.63 (Official Build) nightly (64-bit)
-- | --
Revision | ff5c0da2ec0adeaed5550e6c7e98417dac77d98a-refs/branch-heads/4577@{#1135}
OS | Windows 10 OS Version 2009 (Build 19042.1165)
```

* ensured that `Media Router` is disabled by default via `brave://settings/extensions`
* ensured that `Cast` isn't being displayed under the hamburger menu when  `Media Router` is disabled
* ensured that `Cast` is added into the hamburger menu once `Media Router` is enabled via `brave://settings/extensions`
* ensured that selecting `Cast` opens the `Cast Tab` modal and lists all available devices that can cast
* ensured that casting a `video`, `entire device` & `file` worked as expected

**`Casting from YT`** | **`Same video on TV downstairs`**
-------------------------------|----------------------------------------
casting | Screen Shot 2021-09-11 at 10 12 45 PM

**`Casting video from local file`** | **`Same video on TV downstairs`**
-------------------------------------|---------------------------------------
castingFile | Screen Shot 2021-09-11 at 10 13 19 PM

**`Casting Desktop`**

Screen Shot 2021-09-11 at 10 13 01 PM

**Pull request merged**